### PR TITLE
Validate app name in the deploy form

### PR DESCRIPTION
**Pull request by bryk**
This validates application name asynchronously on the server. Validation is synced with namespace select :)

### CODE REVIEW

**Comment by bryk** [comment]
@zreigz Can you review?

**Comment by zreigz** [comment]
Yes. I am reviewing this

**Comment by zreigz** [comment]
To make it complete you should also check if name contains illegal characters. For example when you enter name: ```some/text``` then you get 500  because ```/``` is forbidden. Also name containing ```%``` returns error. I think it should be done on client side before server side validation.

**Comment by bryk** [comment]
Yes, definitely. But how about doing this in a next pull request? This one is already pretty large and I don't want to make it even larger.

File: src/app/backend/validateappname.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+// limitations under the License.
+
+package main
+
+import (
+	k8serrors "k8s.io/kubernetes/pkg/api/errors"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+type AppNameValiditySpec struct {
```
Comment (zreigz) [inline src/app/backend/validateappname.go]: Add small description

File: src/app/backend/validateappname.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+type AppNameValidity struct {
```
Comment (zreigz) [inline src/app/backend/validateappname.go]: Add small description

**Comment by zreigz** [comment]
Yes I agree it can be done in other PR. Two small remarks and you can merge :)
PTAL

File: src/app/backend/validateappname.go (hunk 2 above)
Comment (bryk) [inline src/app/backend/validateappname.go]: Done.

File: src/app/backend/validateappname.go (hunk 1 above)
Comment (bryk) [inline src/app/backend/validateappname.go]: Done.

**Comment by zreigz** [comment]
LGTM

**Pull request merged**